app/src/test/java/org/jellyfin/androidtv/preference/PreferenceStoreTest.kt

```diff
@@ -0,0 +1,90 @@
+package org.jellyfin.androidtv.preference
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.TestCase
+import io.kotest.matchers.shouldBe
+
+private class TestStub : PreferenceStore() {
+	var key: String? = null
+	private var int: Int? = null
+	private var long: Long? = null
+	private var bool: Boolean? = null
+	private var string: String? = null
+	private var enum: Enum<*>? = null
+
+	@Suppress("UNCHECKED_CAST")
+	override fun <T : Enum<T>> getEnum(preference: Preference<T>): T =
+		(this.enum ?: preference.defaultValue) as T
+
+	override fun getInt(key: String, defaultValue: Int): Int = int ?: 0
+	override fun getLong(key: String, defaultValue: Long): Long = long ?: 0
+	override fun getBool(key: String, defaultValue: Boolean): Boolean = bool ?: false
+	override fun getString(key: String, defaultValue: String) = string ?: ""
+
+	override fun <V : Enum<V>> setEnum(preference: Preference<*>, value: Enum<V>) {
+		key = preference.key
+		enum = value
+	}
+
+	override fun setInt(key: String, value: Int) {
+		this.key = key
+		int = value
+	}
+
+	override fun setLong(key: String, value: Long) {
+		this.key = key
+		long = value
+	}
+
+	override fun setBool(key: String, value: Boolean) {
+		this.key = key
+		bool = value
+	}
+
+	override fun setString(key: String, value: String) {
+		this.key = key
+		string = value
+	}
+
+
+	override fun <T : Any> delete(preference: Preference<T>) {
+		throw NotImplementedError("Not required for tests")
+	}
+}
+
+private enum class TestEnum { NOT_SET, SET }
+
+/**
+ * Tests that the generic set/get methods dispatch to the correct
+ * internal getT/setT methods
+ */
+class PreferenceStoreTest : FunSpec() {
```

Comment:
```suggestion
class PreferenceStoreTest : FunSpec({
```

When doing it like this you don't need the `init {}` block. This is how the other tests work too.

Reply:
It was like this because of the nested class (I didn't even know private classes were a thing). I'll revert it to match the prevailing style :+1: